test/d_do_test.d

```diff
@@ -27,15 +30,16 @@ void usage()
           ~ "   relevant environment variables:\n"
           ~ "      ARGS:          set to execute all combinations of\n"
           ~ "      REQUIRED_ARGS: arguments always passed to the compiler\n"
-          ~ "      DMD:           compiler to use, ex: ../src/dmd\n"
+          ~ "      DMD:           compiler to use, ex: ../src/dmd (required)\n"
           ~ "      CC:            C++ compiler to use, ex: dmc, g++\n"
           ~ "      OS:            win32, win64, linux, freebsd, osx, netbsd\n"
           ~ "      RESULTS_DIR:   base directory for test results\n"
+          ~ "      MODEL:         32 or 64 (required)\n"
           ~ "   windows vs non-windows portability env vars:\n"
           ~ "      DSEP:          \\\\ or /\n"
-          ~ "      SEP:           \\ or /\n"
-          ~ "      OBJ:          .obj or .o\n"
-          ~ "      EXE:          .exe or <null>\n");
+          ~ "      SEP:           \\ or / (required)\n"
+          ~ "      OBJ:          .obj or .o (required)\n"
+          ~ "      EXE:          .exe or <null> (required)\n");
```

Comment:
we could auto-detect the default value here.
For example:

```
version(windows) string exe = ".exe";
else string exe = "";
```

Reply:
Setting reasonable defaults was actually my first thought.  The thinking was it might be nice to make it easier to call `d_do_test.d` on it's own without the Makefile.  However, I realized that `d_do_test.d` is not designed to run on it's own, it's really just an extension for the Makefile, and the reason it's using environment variables in the first place is because the Makefile is not just using those variables to call d_do_test, but also for other parts of it's configuration.

If we wanted to allow the developer to call `d_do_test.d` on it's own I would say let's put defaults in there, but if it's just an extension for the Makefile then I think it makes more sense to just verify the interface between the Makefile and d_do_test (make sure required variables are given).